homework/index.js

```diff
@@ -32,39 +35,59 @@
 
   const root = document.getElementById('root');
   const header = createAndAppend('header', root, { class: 'header' });
-  createAndAppend('p', header, { text: 'HYF Repositories' });
 
-  const select = createAndAppend('select', header, { id: 'selector' });
-  const option = x => {
-    for (let i = 0; i < x.length; i++) {
-      createAndAppend('option', select, { text: x[i].name, value: i });
+  function createHeader() {
+    createAndAppend('p', header, { text: 'HYF Repositories' });
+  }
+  createHeader();
+
+  function createOptions(repositoryList) {
+    const select = createAndAppend('select', header, { id: 'selector' });
+    for (let i = 0; i < repositoryList.length; i++) {
+      const repository = repositoryList[i];
+      createAndAppend('option', select, { text: repository.name, value: i });
     }
-  };
+  }
 
   const mainContainer = createAndAppend('div', root, { id: 'container' });
-  const leftContainer = createAndAppend('div', mainContainer, { class: 'left-block frame' });
-  const table = createAndAppend('table', leftContainer);
-  const listContributor = createAndAppend('tbody', table);
+  function createListContributor() {
+    const leftContainer = createAndAppend('div', mainContainer, { class: 'left-block frame' });
+    const table = createAndAppend('table', leftContainer);
+    const listContributor = createAndAppend('tbody', table);
 
-  const rightContainer = createAndAppend('div', mainContainer, { class: 'right-block frame' });
-  createAndAppend('p', rightContainer, {
-    text: 'Contributions',
-    class: 'contributor',
-  });
+    return listContributor;
+  }
+  const listContributor = createListContributor();
 
-  const list = createAndAppend('ul', rightContainer, { class: 'list1' });
+  function createMainContainer() {
+    const rightContainer = createAndAppend('div', mainContainer, { class: 'right-block frame' });
+    createAndAppend('p', rightContainer, {
+      text: 'Contributions',
+      class: 'contributor',
+    });
+
+    return createAndAppend('ul', rightContainer, { class: 'list1' });
+  }
+  const list = createMainContainer();
```

Comment:
1. This statement should not be at the file level but inside a function body, e.g. `main()`.
2. The name `list` is too generic: list of what? Perhaps `contributorList`?

Reply:
1 is not ok.
2. I did it.